manifests/peers.pp

```diff
@@ -1,11 +1,25 @@
+#
 define haproxy::peers (
   $collect_exported = true,
+  $instance = 'haproxy',
 ) {
 
+  # We derive these settings so that the caller only has to specify $instance.
+  include haproxy::params
+  if $instance == 'haproxy' {
+    $instance_name = 'haproxy'
+    #$config_dir = $haproxy::params::config_dir
+    $config_file = $haproxy::params::config_file
+  } else {
+    $instance_name = "haproxy-${instance}"
+    #$config_dir = inline_template($haproxy::params::config_dir_tmpl)
+    $config_file = inline_template($haproxy::params::config_file_tmpl)
+  }
```

Comment:
i was thinking a function, but i'm not sure… maybe @hunner or @mhaskel or someone else from puppetlabs have an idea?